modal/_container_exec.py

```diff
@@ -16,17 +11,15 @@
 
 from modal_proto import api_pb2
 
-from ._pty import get_pty_info, raw_terminal, set_nonblocking
-from ._utils.async_utils import TaskContext, asyncify
+from ._pty import get_pty_info
 from ._utils.grpc_utils import RETRYABLE_GRPC_STATUS_CODES, retry_transient_errors, unary_stream
+from ._utils.shell_utils import connect_to_terminal, write_to_fd
 from .client import _Client
 from .config import config
-from .exception import ExecutionError, InteractiveTimeoutError, NotFoundError
+from .exception import NotFoundError
 
 
-async def container_exec(
-    task_id: str, command: List[str], *, pty: bool, client: _Client, terminate_container_on_exit: bool = False
-):
+async def container_exec(task_id: str, command: List[str], *, pty: bool, client: _Client):
```

Comment:
We can't remove `terminate_container_on_exit` from the schema yet. Here is the order of roll-out IMO:

- remove from code so clients don't sends terminate_container_on_exit as True anymore
- wait until everyone uses the new client.
- remove the usage of terminate_on_exit from modal-runtime
- remove the schema from client